homework/length-sort/sort.cpp

```diff
@@ -0,0 +1,26 @@
+#include "sort.hpp"
+
+bool cmp(const std::string &a, const std::string &b){
+    //length of both is the same
+    if(a.length() == b.length()){
+        int comparison_result = a.compare(b);
+        if (comparison_result >= 0) // a is greater or identical
+            return true;
+        else
+            return false;            //a is lesser (lexinogr)
+    }
+    if(a.length() > b.length())     //a is longer
+        return true;
+    else                            //a is shorter
+        return false;
+}
+
+std::deque<std::string> lengthSort(std::forward_list<std::string> & words){
+    std::deque<std::string> deq;
+    words.sort(cmp);
+    for (auto word : words){
+        deq.push_front(word);
+    }
+    return deq;
+}
```

Comment:
this is enough to use a simple predicate which checks if both strings are the same length if not compared by length if true compare lexicographically (this is done automatically by using operator <). And insert this predicate to std::sort method.

```
auto pred = [](conat auto& lhs, const auto& rhs) {
   if (lhs.length(0 == rhs.length()) {
       return lhs < rhs;
   }
   
   return lhs.lenght() < rhs.length(); 
}
```